examples/C++/DDS/HelloWorldExampleDiscoveryServer/arg_configuration.h

```diff
@@ -0,0 +1,178 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file arg_configuration.h
+ *
+ */
+
+#ifndef _EPROSIMA_FASTDDS_EXAMPLES_CPP_DDS_HELLOWORLDEXAMPLEDISCOVERYSERVER_ARG_CONFIGURATION_H_
+#define _EPROSIMA_FASTDDS_EXAMPLES_CPP_DDS_HELLOWORLDEXAMPLEDISCOVERYSERVER_ARG_CONFIGURATION_H_
+
+#include "optionparser.h"
+
+#include <iostream>
+#include <regex>
+#include <string>
+
+struct Arg : public option::Arg
+{
+    static void print_error(
+            const char* msg1,
+            const option::Option& opt,
+            const char* msg2)
+    {
+        fprintf(stderr, "%s", msg1);
+        fwrite(opt.name, opt.namelen, 1, stderr);
+        fprintf(stderr, "%s", msg2);
+    }
+
+    static option::ArgStatus Unknown(
+            const option::Option& option,
+            bool msg)
+    {
+        if (msg)
+        {
+            print_error("Unknown option '", option, "'\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+    static option::ArgStatus Required(
+            const option::Option& option,
+            bool msg)
+    {
+        if (option.arg != 0 && option.arg[0] != 0)
+        {
+            return option::ARG_OK;
+        }
+
+        if (msg)
+        {
+            print_error("Option '", option, "' requires an argument\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+    static option::ArgStatus Numeric(
+            const option::Option& option,
+            bool msg)
+    {
+        char* endptr = 0;
+        if (option.arg != 0 && strtol(option.arg, &endptr, 10))
+        {
+        }
+        if (endptr != option.arg && *endptr == 0)
+        {
+            return option::ARG_OK;
+        }
+
+        if (msg)
+        {
+            print_error("Option '", option, "' requires a numeric argument\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+    static option::ArgStatus String(
+            const option::Option& option,
+            bool msg)
+    {
+        if (option.arg != 0)
+        {
+            return option::ARG_OK;
+        }
+        if (msg)
+        {
+            print_error("Option '", option, "' requires a string argument\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+    static option::ArgStatus Locator(
+            const option::Option& option,
+            bool msg)
+    {
+        if (option.arg != 0)
+        {
+            // we must check if it is a correct ip address plus port number
+            if (std::regex_match(option.arg, ipv4))
+            {
+                return option::ARG_OK;
+            }
+        }
+        if (msg)
+        {
+            print_error("Option '", option, "' requires an IPaddress[:portnumber] argument\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+    static const std::regex ipv4;
+};
+
+enum  optionIndex
```

Comment:
I would add new argument to set the **ID** of the server, in the server itself as in the pub & sub.
The way it is right now is simple and fine, but a user that use this example will not be able to work with its own discovery server without that configuration, and it is better to make it visible here.
Use the id as it is used in fastdds cli